### PR TITLE
README for Service Discovery

### PR DESCRIPTION
**Pull request by PettitWesley**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -640,6 +661,125 @@ or
...
 
+#### Using Route53 Service Discovery
+
+With the ECS CLI, you can create a Service that uses [Route53 auto naming for service discovery](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-discovery.html). Service Discovery requires a Service Discovery Service, and a DNS Namespace. When you enable Service Discovery with the ECS CLI, a new Service Discovery Service is always created using CloudFormation. For the DNS Namespace, you have the option of using an existing public or private DNS Namespace, or letting the ECS CLI create a private DNS Namespace for you using CloudFormation. Creation of a Public DNS Namespaces is not supported with the ECS CLI. Please note that only a single DNS Namespace may be used with Service Discovery.
+
+##### Enabling Service Discovery
+
+###### Specifying Values
+
+There are many input parameters for Service Discovery, however, for most of them, the ECS CLI will choose reasonable defaults. Here are the default values and explanations listed with the ECS Params input schema:
```
Comment (SoManyHs) [inline README.md]: I would rephrase this to put a more positive spin -- e.g. "the ECS-CLI simplifies the use of Service Discovery by providing the following defaults, while still allowing maximum configurability..." or something like that.

File: README.md [hunk 2]
```diff
@@ -640,6 +661,125 @@ or
...
+
+```
+version: 1
+run_params:
+  service_discovery:
+    // container name and port are the container that will be used for Service Discovery
```
Comment (SoManyHs) [inline README.md]: Is this comment necessary?

File: README.md [hunk 3]
```diff
@@ -640,6 +661,125 @@ or
...
+    container_name: string            // No Default, required if using SRV records
```
Comment (SoManyHs) [inline README.md]: Not sure you need to say "No default" here and below.

File: README.md [hunk 4]
```diff
@@ -640,6 +661,125 @@ or
...
+    container_port: string            // No Default, required if using SRV records
+    private_dns_namespace:
+      vpc: string                     // Required if ID is not specified
```
Comment (SoManyHs) [inline README.md]: I would say `id` instead of "ID"  to make it obvious you're referring to the field that immediately follows.

File: README.md [hunk 5]
```diff
@@ -640,6 +661,125 @@ or
...
+      id: string                      // Allows you to specify an existing namespace by ID
+      name: string                    // DNS name for the namespace, if it already exists, the CLI will use, if not, the CLI creates one
```
Comment (SoManyHs) [inline README.md]: Possible rephrase: DNS name for private namespace. Used to specify an existing namespace; if one does not exist with this name, the ecs-cli will create it.

File: README.md [hunk 6]
```diff
@@ -640,6 +661,125 @@ or
...
+      description: string             // Only used if the namespace does not yet exist, Default = Created by the Amazon ECS CLI
```
Comment (SoManyHs) [inline README.md]: Period instead of comma, quotes around the default description string.

File: README.md [hunk 7]
```diff
@@ -640,6 +661,125 @@ or
...
+    public_dns_namespace:
+      id: ns-C0VF3F3                  // Specify an existing public namespace by ID
+      name: amazon.com                // Or specify an existing public namespace by Name
```
Comment (SoManyHs) [inline README.md]: I would probably keep these as the type and put the default values in the comment

File: README.md [hunk 8]
```diff
@@ -640,6 +661,125 @@ or
...
+    service_discovery_service:
+      name: string                    // Default = Name of the your ECS Service
```
Comment (SoManyHs) [inline README.md]: Is it obvious what the name of customer's ECS service is?

File: README.md [hunk 9]
```diff
@@ -640,6 +661,125 @@ or
...
+      description: string             // Default = Created by the Amazon ECS CLI
```
Comment (SoManyHs) [inline README.md]: Quotes around default string

File: README.md [hunk 10]
```diff
@@ -640,6 +661,125 @@ or
...
+      dns_config:
+        type: string                  // Valid values: A or SRV, SRV is default/required with network modes bridge and host, A is the default for awsvpc
```
Comment (SoManyHs) [inline README.md]: Period instead of first comma.

Also maybe rephrase the second half to be something like "`SRV` is required/the default when using `bridge` or `host` network mode. `A` is the default for `awsvpc` network mode."

File: README.md [hunk 11]
```diff
@@ -640,6 +661,125 @@ or
...
+        ttl: integer                  // Default = 60
+      healthcheck_custom_config:
+        failure_threshold: integer    // Default = 1
+```
+
+###### Simple Workflow
+
+Let's walkthrough a simple scenario with Service Discovery to see how it works with the ECS CLI. Many of the Service Discovery configuration values can be specified with flags, which take precedence over the ECS Params if both are present. Remember that with the ECS CLI, the Compose Project Name is used as the name for your ECS Service.
```
Comment (SoManyHs) [inline README.md]: walk through

File: README.md (hunk 4 above)
Comment (allisaurus) [reply README.md]: and can you put "id" in quotes, to drive that home?

File: README.md [hunk 12]
```diff
@@ -640,6 +661,125 @@ or
...
+INFO[0066] Updated ECS service successfully              desiredCount=1 serviceName=frontend
+INFO[0081] (service frontend) has started 1 tasks: (task 824b5a76-8f9c-4beb-a64b-6904e320630e).  timestamp="2018-09-12 00:00:26 +0000 UTC"
+INFO[0157] Service status                                desiredCount=1 runningCount=1 serviceName=frontend
+INFO[0157] ECS Service has reached a stable state        desiredCount=1 runningCount=1 serviceName=frontend
+```
+
+Now, the two Services can find each other in the VPC using DNS. The DNS host name will be the name of the Service Discovery Service plus the name of the DNS Namespace. So the ECS Service `frontend` can be found at `frontend.tutorial`, and `backend` can be found at `backend.tutorial`. Remember that since this is a Private DNS Namespace, these domain names can only be resolved within your VPC.
+
+Now, let's update some of the Service Discovery settings for `frontend`; the only values that can be updated are DNS TTL and Health Check Custom Config Failure Threshold (the failure threshold for the health check administered by ECS, which determines when unhealthy containers will have their DNS records removed).
```
Comment (allisaurus) [inline README.md]: can you put the names of the fields that can be updated in `back ticks?`?

File: README.md (hunk 7 above)
Comment (PettitWesley) [reply README.md]: Oh oops those aren't default values those are just bad copy pasta from my design doc...

File: README.md (hunk 8 above)
Comment (PettitWesley) [reply README.md]: In one of the paragraph's below it is stated that this is the Compose Project Name
